dWorldServer/WorldServer.cpp

```diff
@@ -1002,9 +1003,18 @@ void HandlePacket(Packet* packet) {
 					player->GetComponent<CharacterComponent>()->RocketUnEquip(player);
 				}
 
-				c->SetRetroactiveFlags();
-
-				player->RetroactiveVaultSize();
+				// Do charxml fixes here
+				auto* levelComponent = player->GetComponent<LevelProgressionComponent>();
+				if (!levelComponent) return;
+
+				if (levelComponent->GetCharacterVersion() == 1) {
+					Game::logger->Log("WorldServer", "Upgrading Character version to 2");
```

Comment:
What are your thoughts on adding some more information here so we know what character we are upgrading the version of?